servicetalk-buffer-api/src/main/java/io/servicetalk/buffer/api/AsciiBuffer.java

```diff
@@ -114,9 +114,16 @@ Buffer unwrap() {
      * @throws NullPointerException if {@code subString} is {@code null}.
      */
     int indexOf(char ch, int start) {
+        if (!singleByte(ch)) {
+            return -1;
+        }
         return buffer.indexOf(start, buffer.writerIndex(), (byte) ch);
```

Comment:
nit: simplify and remove negation: `return singleByte(ch) ? buffer.indexOf(start, buffer.writerIndex(), (byte) ch) : -1;`